python/python/ert_gui/simulation/detailed_progress.py

```diff
@@ -166,6 +189,73 @@ def headerData(self, index, orientation, role=Qt.DisplayRole):
         if orientation == Qt.Vertical:
             return QVariant(index)
 
+class SingleTableView(QTableView):
+    def __init__(self):
+        super(SingleTableView, self).__init__()
+        self.open_files = {}
+        self.realization = -1
+        self.iteration = -1
+
+    def mousePressEvent(self, mouse_event):
+        pos = mouse_event.pos()
+        index = self.indexAt(pos)
+        selected_file = self.model().get_file_name(index)
+        if selected_file and not selected_file in self.open_files:
+            job_name = self.model().model_data[index.row()][self.model().get_column_index("name")]
+            viewer = FileViewer(self, selected_file, job_name, index.row(), self.realization, self.iteration)
+            self.open_files[selected_file] = viewer
+
+        elif selected_file in self.open_files:
+            self.open_files[selected_file].reload(selected_file)
+
+    def update_data(self, jobs, iteration, realization):
+        self.setMinimumHeight(200)
+        self.iteration = iteration
+        self.realization = realization
+        model_data = []
+        headers = []
+        for job in jobs:
+            data = job.dump_data()
+            row = [str(data[key]) for key in data]
+            model_data.append(row)
+            headers = list(data.keys())
+
+        self.model().update_data(headers, model_data)
+        self.resizeColumnsToContents()
+        self.model().modelReset.emit()
+
+        for file_name in self.open_files:
+            if self.open_files[file_name].isVisible():
+                self.open_files[file_name].reload(file_name)
+
+
+class FileViewer(QDialog):
+    def __init__(self, parent, file_name, job_name, job_number, iteration, realization):
+        super(FileViewer, self).__init__(parent)
+
+        self.setWindowTitle("{} # {} Realization: {} Iteration: {}" \
+                            .format(job_name, job_number, realization, iteration))
+
+        self.text_cont = QTextEdit()
+        self.text_cont.setReadOnly(True)
+        close_button = QPushButton("Close")
+        close_button.clicked.connect(self.accept)
+
+        layout = QGridLayout(self)
```

Comment:
Grid layouts should be used exclusively when you have a grid (remember the early html days where you had recursive `<table>` 27 levels deep?)
I'd use a `QVBoxLayout` here, although the button probably has a fixed height, so it probably won't make any visual difference